pkg/infra/baseagent.go

```diff
@@ -85,11 +87,21 @@ func (a *Base) updateHP(foodTaken float64) {
 	a.hp = int(math.Min(100, float64(a.hp)+foodTaken))
 }
 
+func (a *Base) HasEaten() bool {
+	return a.hasEaten
+}
+
+func (a *Base) setHasEaten(newStatus bool) {
+	a.hasEaten = newStatus
+}
+
 func (a *Base) TakeFood(amountOfFood float64) float64 {
-	if a.floor == a.tower.currPlatFloor {
+	if a.floor == a.tower.currPlatFloor && !a.HasEaten() {
```

Comment:
```suggestion
	if a.floor == a.tower.currPlatFloor && !a.hasEaten {
```
Custom agent can implement `HasEaten()` and not sure what the behavior would be